ui/src/ifql/ast/walker.ts

```diff
@@ -238,15 +238,42 @@ export default class Walker {
     })
   }
 
+  private constructObject(value) {
+    const propArray = _.get(value, 'properties', [])
+    const valueObj = propArray.reduce((acc, p) => {
+      return {...acc, [p.key.name]: p.value.name}
+    }, {})
+    return valueObj
+  }
+
+  private constructArray(value) {
+    const elementsArray = _.get(value, 'elements', [])
+    const valueArray = elementsArray.reduce((acc, e) => {
+      return [...acc, e.value]
+    }, [])
+    return valueArray
+  }
+
   private getProperties = props => {
-    return props.map(prop => ({
-      key: prop.key.name,
-      value: _.get(
-        prop,
-        'value.value',
-        _.get(prop, 'value.location.source', '')
-      ),
-    }))
+    return props.map(prop => {
+      const key = prop.key.name
+      let value
+      if (_.get(prop, 'value.type', '') === 'ObjectExpression') {
```

Comment:
How do you feel about making an enum or constant  that includes `ObjectExpression` and `ArrayExpression` to avoid having hard coded strings?